compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -916,8 +916,7 @@ object RefChecks {
      *   module that declares `sym`.
      */
     def skipWarning(using Context) =
-      if sym.isEnum then ctx.owner.ownersIterator.exists(isDeprecatedOrEnum)
-      else ctx.owner.ownersIterator.exists(isDeprecatedOrSyntheticMethod)
+      ctx.owner.ownersIterator.exists(if sym.isEnumCase then isDeprecatedOrEnum else isDeprecatedOrSyntheticMethod)
```

Comment:
`isDeprecatedOrEnum` still needs to call `isDeprecatedOrSyntheticMethod`, the condition here was just to lift the check out of the loop